tt_metal/impl/device/device_pool.hpp

```diff
@@ -51,6 +57,12 @@ class DevicePool {
     std::vector<uint32_t> l1_bank_remap;
     std::mutex lock;
     std::vector<std::unique_ptr<Device>> devices;
+    // Used to track worker thread handles (1 worker thread created per device)
+    // when we need to check if a call is made from an application thread or a
+    // worker thread
+    std::unordered_map<Device*, std::thread::id> worker_thread_ids_per_device;
```

Comment:
minor thing but the name is a misnomer. maybe `device_to_worker_thread` is a better name?

Reply:
Updated to `device_to_worker_thread_id`

Reply:
great!